controllers/admin_domains.php

```diff
@@ -543,9 +541,374 @@ public function importPackages()
+            foreach ($currencies as $currency) {
+                for ($i = 1; $i <= 10; $i++) {
+                    foreach ($configurable_option->values as &$value) {
+                        // Check if the term already exists
+                        $exists_pricing = false;
+                        foreach ($value->pricing as &$pricing) {
+                            if ($pricing->term == $i && $pricing->period == 'year' && $pricing->currency == $currency->code) {
+                                $exists_pricing = true;
+                            }
+                        }
+
+                        // If the pricing not exists, add a placeholder for that pricing
+                        if (!$exists_pricing) {
+                            $value->pricing[] = (object)[
+                                'term' => $i,
+                                'period' => 'year',
+                                'currency' => $currency->code
+                            ];
+                        }
+                    }
+                }
+            }
+
+            // Order pricing rows
+            foreach ($configurable_option->values as &$value) {
+                usort($value->pricing, function ($a, $b) {
+                    if ($a->term == $b->term) {
+                        return 0;
+                    }
+
+                    if ($a->term < $b->term) {
+                        return -1;
+                    }
+
+                    return 1;
+                });
+            }
+        } catch (Throwable $e) {
+            echo $this->setMessage(
+                'error',
+                ['exception' => [$e->getMessage()]],
+                true,
+                ['show_close' => false],
+                false
+            );
+
+            return false;
+        }
+
+        echo $this->partial(
+            'admin_domains_configurableoptions_pricing',
+            compact(
+                'configurable_option',
+                'currencies',
+                'default_currency'
+            )
+        );
+
+        return false;
+    }
+
+    /**
+     * Updates the pricing of a configurable option
+     */
+    public function updateConfigurableOption()
+    {
+        $this->uses(['PackageOptions', 'Currencies']);
+        $this->components(['Record']);
+
+        // Fetch the configurable option
+        if (
+            !$this->isAjax()
+            || !isset($this->get[0])
+            || !($option = $this->PackageOptions->get($this->get[0]))
+        ) {
+            //$this->redirect($this->base_uri . 'plugin/domains/admin_domains/configurableoptions/');
+        }
+
+        if (!empty($this->post['pricing'])) {
+            // Build values array
+            $value_vars = [];
+            foreach ($this->post['pricing'] as $value_id => $pricing) {
+                // Get value
+                $value = $this->Record->select(['package_option_values.*', 'package_options.type' => 'option_type'])
```

Comment:
Since the $value_id is from user input, there is no guarantee it is valid.  We should either give an error or skip the $value_id.